src/inject/dynamic-theme/css-rules.ts

```diff
@@ -41,23 +42,33 @@ function isCSSVariable(property: string) {
     return property.startsWith('--') && !property.startsWith('--darkreader');
 }
 
+function getParentGroups(rule: CSSRule, stack: string[] = []): string[] {
+    const {parentRule} = rule;
+    if (!parentRule || !(parentRule instanceof CSSMediaRule)) {
+        return stack;
+    }
+    stack.push(parentRule.conditionText);
+    return getParentGroups(parentRule, stack);
+}
+
 export function getCSSVariables(rules: CSSRuleList) {
-    const variables = new Map<string, string>();
+    const variables = new Map<string, Variable>();
     rules && iterateCSSRules(rules, (rule) => {
+        const parentGroups = getParentGroups(rule);
```

Comment:
The result should be the same for each sibling. Try optimizing it. E.g. use WeakMap to cache results for each `parentRule`, but only in `getCSSVariables` function scope to prevent unpredictable bugs.

Maybe `iterateCSSRules` is not perfect for iterating trees and should be refactored in future.